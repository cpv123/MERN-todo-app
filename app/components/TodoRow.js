import React, { Component } from 'react';
import { ListItem } from 'material-ui/List';
import Checkbox from 'material-ui/Checkbox';
import FontIcon from 'material-ui/FontIcon';
import IconButton from 'material-ui/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import Popover from '@material-ui/core/Popover';
import Paper from 'material-ui/Paper';
import MoreIcon from '@material-ui/icons/More';

export default class TodoRow extends Component {
  constructor(props) {
    super(props)
    this.state = {
      anchorEl: null 
    }
  };

  onClickDelete = event => {
    this.setState({
      anchorEl: event.currentTarget,
    });
  };

  onCloseDelete = () => {
    console.log('closing')
    this.setState({
      anchorEl: null,
    });
  };

  render() {
    const { todo, tags, complete } = this.props;

    return (
      <div>
      <ListItem
        primaryText={todo}
        leftCheckbox={
          <Checkbox
            checked={complete ? true : false}
          />
        }
        rightIconButton={
          <IconButton 
            onClick={this.onClickDelete} >
            <DeleteIcon />
            <MoreIcon />
            <Popover
              open={Boolean(this.state.anchorEl)}
              anchorEl={this.state.anchorEl}
              onClose={this.onClose}
              anchorOrigin={{
                vertical: 'top',
                horizontal: 'left',
              }}
              transformOrigin={{
                vertical: 'bottom',
                horizontal: 'left',
              }}
            >
              <Paper style={{ 'padding': 15 }}>Delete?</Paper>
            </Popover>
          </IconButton>
        }
        style={complete ? styles.completed : {}}
      />
      </div>
    );
  }
}

const styles = {
    completed: {
        color: 'gray',
        textDecoration: 'line-through'
    }
};