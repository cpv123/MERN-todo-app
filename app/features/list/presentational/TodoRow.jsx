import React from 'react';
import { ListItem } from 'material-ui/List';
import Checkbox from 'material-ui/Checkbox';
import IconButton from 'material-ui/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';

const styles = {
  completed: {
    color: 'gray',
    textDecoration: 'line-through',
  },
};

export default function TodoRow({
  todo, complete, onClickComplete, onClickDelete,
}) {
  return (
    <div style={{ width: '80%' }}>
      <ListItem
        primaryText={todo}
        leftCheckbox={(
          <Checkbox
            onCheck={onClickComplete}
            checked={!!complete}
          />
        )}
        rightIconButton={(
          <IconButton
            onClick={onClickDelete}
          >
            <DeleteIcon />
          </IconButton>
        )}
        style={complete ? styles.completed : {}}
      />
    </div>
  );
}
