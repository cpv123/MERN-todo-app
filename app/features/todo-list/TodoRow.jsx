import React from 'react';
import PropTypes from 'prop-types';
import { ListItem } from 'material-ui/List';
import Checkbox from 'material-ui/Checkbox';
import IconButton from 'material-ui/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';

const styles = {
  completed: {
    color: 'gray',
    textDecoration: 'line-through',
  },
};

export default function TodoRow({
  todo, complete, onClickComplete, onClickDelete,
}) {
  return (
    <div style={{ width: '80%' }}>
      <ListItem
        primaryText={todo}
        leftCheckbox={(
          <Checkbox
            onCheck={onClickComplete}
            checked={!!complete}
          />
        )}
        rightIconButton={(
          <IconButton
            onClick={onClickDelete}
          >
            <DeleteIcon />
          </IconButton>
        )}
        style={complete ? styles.completed : {}}
      />
    </div>
  );
}

TodoRow.propTypes = {
  todo: PropTypes.string.isRequired,
  complete: PropTypes.bool,
  onClickComplete: PropTypes.func.isRequired,
  onClickDelete: PropTypes.func.isRequired,
};

TodoRow.defaultProps = {
  complete: false,
};
