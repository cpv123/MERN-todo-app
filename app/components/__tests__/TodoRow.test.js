/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import TodoRow from '../TodoRow.jsx';

describe('TodoRow', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TodoRow />);
    expect(wrapper).toMatchSnapshot();
  });
});