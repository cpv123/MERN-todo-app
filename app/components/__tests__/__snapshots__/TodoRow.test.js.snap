// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoRow renders correctly 1`] = `
<div
  style={
    Object {
      "width": "80%",
    }
  }
>
  <ListItem
    autoGenerateNestedIndicator={true}
    containerElement="span"
    disableKeyboardFocus={false}
    disabled={false}
    initiallyOpen={false}
    insetChildren={false}
    leftCheckbox={
      <Checkbox
        checked={false}
        disabled={false}
        labelPosition="right"
      />
    }
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    primaryTogglesNestedList={false}
    rightIconButton={
      <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <pure(Delete) />
      </IconButton>
    }
    secondaryTextLines={1}
    style={Object {}}
  />
</div>
`;
